src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -67,1008 +50,333 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
-    * @param addr1 an Address
-    * @param addr2 an Address
-    * @return boolean. true if the seasonal fields of the two addresses overlap
+    * @description helper class to keep track of the default and seasonal address for a household
     */ 
-    public static boolean isSeasonalOverlap(Address__c addr1, Address__c addr2) {
-        if (addr1 == null || addr2 == null) 
-            return false;        
-        if (!isSeasonalAddr(addr1) || !isSeasonalAddr(addr2))
-            return false;
-
-		
-		//If only one of the addresses has start and end year but the other one doesn't, we can safely remove the 
-        //start and end year from the one that has them for the seasonal overlap comparison (we do not accept seasonal 
-        //addresses with only start or only end year). The one that doesn't have start and end year is valid every single
-        //year.
-        Integer startYear1 = (addr1.Seasonal_Start_Year__c == null ? System.today().year() : Integer.valueOf(addr1.Seasonal_Start_Year__c));
-        Integer endYear1 = (addr1.Seasonal_End_Year__c == null ? System.today().year() : Integer.valueOf(addr1.Seasonal_End_Year__c));
-        Integer startYear2 = (addr2.Seasonal_Start_Year__c == null ? System.today().year() : Integer.valueOf(addr2.Seasonal_Start_Year__c));
-        Integer endYear2 = (addr2.Seasonal_End_Year__c == null ? System.today().year() : Integer.valueOf(addr2.Seasonal_End_Year__c));
-        
-        Date start1 = Date.newInstance(startYear1, Integer.valueOf(addr1.Seasonal_Start_Month__c), Integer.valueOf(addr1.Seasonal_Start_Day__c));
-        Date end1 = Date.newInstance(endYear1, Integer.valueOf(addr1.Seasonal_End_Month__c), Integer.valueOf(addr1.Seasonal_End_Day__c));
-        Date start2 = Date.newInstance(startYear2, Integer.valueOf(addr2.Seasonal_Start_Month__c), Integer.valueOf(addr2.Seasonal_Start_Day__c));
-        Date end2 = Date.newInstance(endYear2, Integer.valueOf(addr2.Seasonal_End_Month__c), Integer.valueOf(addr2.Seasonal_End_Day__c));
-         
-        UTIL_Debug.debug('****Start date 1: ' + start1);
-        UTIL_Debug.debug('****End date 1: ' + end1);
-        UTIL_Debug.debug('****Start date 2 : ' + start2);
-        UTIL_Debug.debug('****End date 2: ' + end2);
-        
-        Boolean isOverlap = false;
-        //   date1 start contained in date2     -   date2 start contained in date1
-		if ((start1 >= start2 && start1 <= end2) || (start2 >= start1 && start2 <= end1) 
-		//  date2 end contained in date1, overlapping year - date1 end contained in date2, overlapping year 
-		|| (end2 < start2 && end2 >= start1) || (end1 < start1 && end1 >= start2)
-		// both date1 and date2 overlap the year, and one of them overlaps the other
-		|| (end1 < start1 && end2 < start2 && (end2 >= end1 && end2 <= start1 || end1 >= end2 && end1 <= start2))) {
-		    isOverlap = true;
-		}
-        
-        return isOverlap;
+    public class ParentInfo {
```

Comment:
Classes must use `with sharing` to enforce record level access and avoid bypassing record level security